include/silo/storage/column/insertion_index.h

```diff
@@ -0,0 +1,74 @@
+#ifndef SILO_INSERTION_INDEX_H
+#define SILO_INSERTION_INDEX_H
+
+#include <array>
+#include <memory>
+#include <regex>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+#include <roaring/roaring.hh>
+
+#include "silo/common/nucleotide_symbols.h"
+
+namespace boost::serialization {
+struct access;
+}  // namespace boost::serialization
+
+namespace silo::storage::column::insertion {
+
+class InsertionIndex {
+   friend class boost::serialization::access;
+
+  public:
+   using three_mer_t = std::array<size_t, 3>;
+   using sequence_ids_t = std::vector<uint32_t>;
+
+  private:
+   template <class Archive>
+   [[maybe_unused]] void serialize(Archive& archive, const uint32_t /* version */) {
+      // clang-format off
+      archive& insertion_positions;
+      // clang-format on
+   }
+
+   struct Insertion {
+      std::string value;
+      sequence_ids_t sequence_ids;
+   };
+
+   using one_mer_index_t = std::array<sequence_ids_t, NUC_SYMBOL_COUNT>;
```

Comment:
I was a little confusing that the typedef `sequence_ids_t` was used here. `insertion_ids_t` would be a bit more descriptive I would think